### DISCUSSION TITLE
Master - detail drop down listbox (for example Country - City)

### DISCUSSION THREAD

**Discussion by atamersignum** [Budibase/budibase#3983] [Help]
Hello,
I have a budibase form which  include country and city listboxes. I want  the city list is filtered when I select a country from the country list.
Can anybody help me?
Thank you

**Comment by mjashanks** [comment]
Hi @atamersignum 

To do this, follow steps below (also see my video)
1. Create a table called "Cities", with column `City` (Text) and `Country` (Text)
2. Create a view called "Countries", on your table with `Calculate > Count of City` and `Group by > Country`. This produces a distinct list of cities.
3. On your form, Add a data provider, with data source set the data source to your "Countries" view
4. Underneath, add an options picker
-  set the field name. You can free type in here is you don't have a `Country` column
- Set the options source to `Data Provider`
- Set the data provider to your provider from step 3
- Set both the `Label` and `Value` to `group` ("Group" is the "Group By" name of the view i.e. the "Country" column)
5. Add another Data provider, linked to the "Cities" table. 
6. Click "Define Filters" and set `where Country Equals Binding {{ Form.Fields.Country }}` ...i.e. filter cities by the selected country
7. Add another options picker
- Set the field to your City column
- Set the options source to data provider
- set the data provider to your provider from step 6
- Set both the `Value` and `Label` to "City" (name of the city)
That should be it!

https://user-images.githubusercontent.com/3524181/148953566-fa461273-bbdd-4e49-9a9d-c14406aba480.mp4

**Reply by edrisi6** [reply]
Absolutely great help. Thank you

**Comment by atamersignum** [comment]
Thank you very much.

**Comment by atamersignum** [comment]
I got positive result. I tried by creating relationship type columns but could not get any results. Thank you very much

**Comment by leonlazic** [comment]
I cannot find a group by function in my view. I am using Budibase's internal database.

**Reply by ConorWebb96** [reply]
Hey @leonlazic,

[Heres](https://github.com/Budibase/budibase/discussions/3933#discussioncomment-7454561) the explanation for the group by feature you can't find.
